src/main/java/teammates/ui/webapi/output/InstructorData.java

```diff
@@ -0,0 +1,20 @@
+package teammates.ui.webapi.output;
+
+import java.util.Optional;
+
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+
+/**
+ * The API output format of {@link InstructorAttributes}.
+ */
+public class InstructorData extends ApiOutput {
+    private final String name;
+
+    public InstructorData(InstructorAttributes instructorAttributes) {
```

Comment:
In hindsight, perhaps it was a bad idea to use `InstructorAttributes` to construct this object.

Instead, I suggest you rewrite the constructor to `public InstructorData(String instructorName)` and set the name directly. This will require the action to be updated too. The responsibility of the null check will then be left to the action (which it already is, I missed it on my first review).

My reasoning:
- By using `InstructorAttributes` directly, we need to put a null check + default value into this DTO. This can be considered business logic, which does not belong in a _response_ DTO. The responsibility of forming a valid response should be left to the action.
- We only need the name of the instructor for this response. By using `instructorName` directly, there is little to no ambiguity as to what is necessary to create this DTO.
    - If you use `InstructorAttributes`, you need to click in to see what is actually being used, which is annoying.
    - It is also dishonest, because the dependencies between this DTO and `InstructorAttributes` are unclear and are not strong enough to warrant using the `InstructorAttributes` object.

Reply:
Yeap it makes sense! I suspected we may need more data from this endpoint in the future so just used the entire `InstructorAttributes` object. However, that is not correct; we should not couple them this way. 

Changed!